addons/web/static/src/views/relational_model.js

```diff
@@ -1720,19 +1742,9 @@ export class StaticList extends DataPoint {
                 return record;
             })
         );
-        if (!hasSeveralPages && orderFieldName) {
+        if (!hasSeveralPages && orderFieldNames.length) {
             this.records.sort((r1, r2) => {
-                let v1 = r1.data[orderFieldName];
-                let v2 = r2.data[orderFieldName];
-                if (this.fields[orderFieldName].type === "many2one") {
-                    v1 = v1[1];
-                    v2 = v2[1];
-                }
-                if (v1 <= v2) {
-                    return this.orderBy.asc ? -1 : 1;
-                } else {
-                    return this.orderBy.asc ? 1 : -1;
-                }
+                return compareRecords(r1.data, r2.data);
```

Comment:
don't we need to make sure that orderFieldNames are fetched in this case?

Can we sort by x2many fields?